crates/usvg/src/text/flatten.rs

```diff
@@ -19,58 +24,134 @@ fn resolve_rendering_mode(text: &Text) -> ShapeRendering {
             let mut path = path.clone();
             path.rendering_mode = rendering_mode;
-            new_paths.push(path);
+            new_children.push(Node::Path(Box::new(path)));
         }
 
+        // Instead of always processing each glyph separately, we always collect
+        // as many outline glyphs as possible by pushing them into the span_builder
+        // and only if we encounter a different glyph, or we reach the very end of the
+        // span to we push the actual outline paths into new_children. This way, we don't need
+        // to create a new path for every glyph if we have many consecutive glyphs
+        // with just outlines (which is the most common case).
         let mut span_builder = tiny_skia_path::PathBuilder::new();
 
         for glyph in &span.positioned_glyphs {
-            if let Some(outline) = fontdb.outline(glyph.font, glyph.glyph_id) {
-                if let Some(outline) = outline.transform(glyph.transform) {
-                    span_builder.push_path(&outline);
+            // A COLRv0 glyph. Will return a vector of paths that make up the glyph description.
+            // TODO: Don't use black for foreground color? But not sure whether to use fill or stroke
+            // color.
+            if let Some(layers) = fontdb.colr(glyph.font, glyph.id, Color::black()) {
+                push_outline_paths(span, &mut span_builder, &mut new_children, rendering_mode);
+
+                let mut group = Group {
+                    transform: glyph.colr_transform(),
+                    ..Group::empty()
+                };
+
+                for path in layers {
+                    // TODO: Probably need to update abs_transform of children?
+                    group.children.push(Node::Path(Box::new(path)));
                 }
+                group.calculate_bounding_boxes();
+
+                new_children.push(Node::Group(Box::new(group)));
             }
-        }
+            // An SVG glyph. Will return the usvg tree containing the glyph descriptions.
+            else if let Some(tree) = fontdb.svg(glyph.font, glyph.id) {
+                push_outline_paths(span, &mut span_builder, &mut new_children, rendering_mode);
 
-        if let Some(path) = span_builder.finish().and_then(|p| {
-            Path::new(
-                String::new(),
-                span.visibility,
-                span.fill.clone(),
-                span.stroke.clone(),
-                span.paint_order,
-                rendering_mode,
-                Arc::new(p),
-                Transform::default(),
-            )
-        }) {
-            stroke_bbox = stroke_bbox.expand(path.stroke_bounding_box());
-            new_paths.push(path);
+                let mut group = Group {
+                    transform: glyph.svg_transform(),
+                    ..Group::empty()
+                };
+                // TODO: Probably need to update abs_transform of children?
```

Comment:
Yeah, I'm still thinking about a better `abs_transform` implementation. The current one is pretty meh.